### ISSUE TITLE
Nested Folders

### ISSUE THREAD

**Issue by steviehs** [scubajeff/lespas#28]
On my Computer, I organize my pictures in nested folders like "2021/10" "2021/09" and so on. (in fact I gave up organizing by event :-)  While converting this to my mobile album, I create a flat folder hierarchy like 2021_10, 2021_09. This list gets quite long after 30 years of collecting pictures. Is there a reason why Les Pas - like most of the other picture viewers on Android - does not allow nested folders?

**Comment by scubajeff** [comment]
Almost every request for nested folder support is about grouping photos by months, years etc. To me, this shouldn't be done by grouping but rather searching. There is already a search entry in Les Pas, although what you can do now is AI search for certain types of object, but it should be easy to add a search by date feature there too.

**Comment by steviehs** [comment]
Right now I use f-stop as my regular gallery, the main reason is speed, it is really fast. LesPas is regarding speed in the same league. f-stop does also not support nested folders, so I had to flatten my folder hierarchy for the phone gallery...
Still I would also think nested folders would be nice. Searching for something instead of browsing for something is quite different from the intuitive view, I'd say...

**Comment by nicuh** [comment]
Hi @scubajeff, I just discovered your app and it looks very promising.
But I have to agree with the above comments regarding the lack of nested folders.

For me, flat folders will not work because I have many pictures that I already organized like this:

Events 
  - Birthdays
      - Wife
      - Kids
 Vacation
  - Place a
  - Place b
 People
  - Person a
  - Person b

Tags might address this, but I see from the FAQ it is not supported because you don't want to use tags from Nextcloud. I understand, it makes sense, but you don't have to: EXIF already has support for tags/keywords.

This is a show stopper for me. Actually, I might be able to get used to it, but I will never be able to convince my wife and parents to use it.

Please reconsider this feature because, as I said, the app is promising: I like the fact that I can have photos on my phone but at the same time they are 2-way synced with Nextcloud. I could not find this feature in any other mobile apps. The fact that I can even have photos remotely, but still see them in the app is a bonus.

**Comment by scubajeff** [comment]
@nicuh Yes, EXIF could be a good candidate for tagging photo, it's what it means for and many vendors do this. The obstacle Les Pas facing when adapting it is, original full resolution picture file is needed on the phone so that it's EXIF portion can be edited. This means for the simple job as adding a tag to a picture, several mega bytes of data have to travel back and forth between your phone and the Nextcloud server. Considering tagging serveral hundreds of them...

One way to handle this is a companion app running on server side to do the tagging job locally. Then the major problem left is the 2 phase commitment of the new file on both side. But sadly I don't have proper skill of PHP, so let's hope for some PHP guys can join this project.

**Comment by scubajeff** [comment]
It's easy to understand nested folder structure when you are working with the file system tree, but Les Pas is a pure mobile experience, a sub-album within an album is quite unnatural, also bring challenge to UI and UX design.

**Comment by Universal-Igloo** [comment]
Ah, I just posted a [similar issue](https://github.com/scubajeff/lespas/issues/153) , sorry didn't see this earlier. But that one is still relevant given the type of implementation requested.

While I understand that les pas is a pure mobile experience, the purpose of lespas for us is to backup photos AND then down the line be able to easily find them on the computer I sync my server with or on the nextcloud web app.

Hence, I strongly believe this option is much needed for just the "Backup" archive folder.

Also, in the official nextcloud app, using Year/Month subfolders is a **configurable option**. So, not everyone has to agree with it, but the option exists for those it would help!

Like some others above, it is not possible for me to use lespas without this.

**Comment by allista** [comment]
Despite your strong opinion on the matter, @scubajeff , here's another example of nesting/tagging (better both) as the only means of managing a big collection accumulated through the years:

~350GB
~70k photos

Current structure:

- People/
    - Ann/
        - Birthday-2009/
            - raw/
                - original-1234.nef
                - ... 
            - processed-selected-1234.jpeg 
            - ...
        - Hiking-2014/
            - raw/
                - original-2345.nef 
            - processed-selected-2345.jpeg 
            - ...
        - PhotoSession-2016/
        - etc...
    - Kate/
    - Tom/
    - Mom/
    - Dad/
    - Kid/
    - ...another 10 people
- NewYears/
    - 2007/
    - 2009-at-school/
    - 2014-trip-to-Sydney/
    - ... 
- Landscape/
    - 2017-09-12/
        - raw/
        - ... 
    - 2017-10-23/
    - 2018-03-15/
    - ...
- and so on, and on

More than 8000 nested folders in total :warning: 

Now, I can't imagine myself scrolling through 8k flat albums, even though each of them does indeed represent an event. But I would very much like the ability to browse the collection with my phone, processing selected images on the go with Snapseed.

And considering that the storage backend you use (nextcloud) natively supports nested folders, the limitation of Les Pas in this respect _seems_ a bit artificial :shrug:

**Comment by jwalt** [comment]
I‌ want to add another important use-case for subfolders: Sharing. I‌ group my pictures by topics like several people have said, although just two levels (Category/Event, e.g. "Family/Reunion 2023").

With the current flat structure, I cannot easily share all pictures of a certain hobby on the nextcloud server with friends from that hobby. It's either giving them access to all pictures (a no-go), or creating 20 individual shares, and remembering to share each new subfolder, and keeping all of them in sync -- this is not practical in any way.

> It's easy to understand nested folder structure when you are working with the file system tree, but Les Pas is a pure mobile experience, a sub-album within an album is quite unnatural, also brings challenge to the design of both UI and UX.

While I do not have 8000 folders, scrolling through 80 folders accumulated over 5 years  is already painful, this is _not_ mobile friendly.

Lespas already has a hierarchical view when you enable the phone gallery. I think that would be a good starting point for a UI. Even just two levels would make a huge difference. Please reconsider your opinion.

**Comment by scubajeff** [comment]
As you might already point out, a new set of concept has to be introduced to support hierachy/nested folder, even if it's just two levels. This concept, let's just call it 'Group', it's on top of albums. So you have several groups and each group has several albums in it. It's pretty obviouse and understandable if you already organized your collection in such a way. But considering the other case of another type of new user, who has no idea of all these tricks. When he/she tries to create his/her first album by adding a picture to Les Pas, there would be a little frustration ahead of him/her, e.g., he/she need to create a group first, which I believe is a concept hard to swallow for this user. 

Don't get me wrong, nested folder is a great way for photo collection. It's just too difficult to impletment because there are a lot of things to consider and balance, concept naming, UI simplicity, genuine UX flow, just to name a few.

For Gallery, Les Pas acts as a viewer only, that's not the same case and it's easy.

**Comment by AntoineHX** [comment]
I'd also welcome a nested folder feature.
I can understand that fully support this feature on the mobile side is too difficult/time consuming. 
Maybe a good starting point would be for Les Pas to support a nested architecture as a viewer only, in the same way it's already done for the photo gallery (like @jwalt mentioned). Allowing to sort photo view of an album by folder rather than ignoring completely the nested folder (That's the behaviour I've noticed on my side). 
It would still require to create nested folder on the server/computer side but at least it wouldn't force to use a flat architecture like @allista , @nicuh , @Forza-tng and @steviehs mentioned.

**Comment by jwalt** [comment]
> As you might already point out, a new set of concept has to be introduced to support hierachy/nested folder, even if it's just two levels. This concept, let's just call it 'Group', it's on top of albums. So you have several groups and each group has several albums in it. It's pretty obviouse and understandable if you already organized your collection in such a way. But considering the other case of another type of new user, who has no idea of all these tricks. When he/she tries to create his/her first album by adding a picture to Les Pas, there would be a little frustration ahead of him/her, e.g., he/she need to create a group first, which I believe is a concept hard to swallow for this user.

Yeah, I am totally on board with new-user compatibility. I like the idea of calling this "Groups", and there can be a default/unnamed group that represents the toplevel folder. So for new users, there would only be this one unnamed group that is always there. No change in workflow to the current state. The menu could have a "Create Album Group" entry, and only then the UI gets slightly more complex.

> Don't get me wrong, nested folder is a great way for photo collection. It's just too difficult to impletment because there are a lot of things to consider and balance, concept naming, UI simplicity, genuine UX flow, just to name a few.

Yes, there would be some UX to be defined, but I don't think this would be too disruptive. The top-level album-list is the most visible change, everything else could be hidden in a menu: create group (i.e. folder), remove group (only if group is empty), move album to different group. Those three operations should be a viable minimal version of that feature. Anything else (e.g. album lists for moving/copying images) can stay as it is now, just with albums named `group-name/album-name`. Even that minimal feature would make working with 80+ albums much, much easier.
